dags/veda_data_pipeline/utils/collection_generation.py

```diff
@@ -0,0 +1,135 @@
+from typing import Any, Dict
+
+import fsspec
+import xarray as xr
+import xstac
+from veda_data_pipeline.utils.schemas import SpatioTemporalExtent
+
+
+class GenerateCollection:
+    common_fields = [
+        "title",
+        "description",
+        "license",
+        "links",
+        "time_density",
+        "is_periodic",
```

Comment:
by using a custom model, will other fields be stripped from the input if not included here? Or will these be dropped by the custom schema anyway? (please disregard if this is something we have to live with until we transition to pystac).

`[ renders, stac_extensions, assets, item_assets, summaries, and providers]` are all valid fields that could be provided for a new dataset

Reply:
Isayah and I talked this over, we added these to the common_fields but don't necessarily think we need to add in validation for these fields in this PR. Such as the renders stac_extension should be listed if the fields are present, would be something handled in a different PR or service

I think there could potentially be a larger list if we start throwing in all the stac_extensions?

Reply:
> don't necessarily think we need to add in validation for these fields in this PR

100% agree, I am just hoping to prevent stripping valid information that an operator might have put in the collection record (people are surprised when they take the time to write renders metadata only to see it has disappeared in the stac api after running the old datasets endpoint). I will be curious to see if adding in these fields will allow users to submit more detailed collection metadata.

> I think there could potentially be a larger list if we start throwing in all the stac_extensions?

Also agree, I think eventually we need to be more flexible and use something like the pystac Collection instead of hard coded pydantic models.

Reply:
I guess it will now be possible for someone to submit invalid metadata in a collection as in `providers: None`.

The datasets endpoint is a can of 🪱 s; sorry I nitpicked it.

Reply:
I like the idea of pystac collection as a validation method for collections. Could be something to explore. @jjfrench if you want to set up a meeting we can all discuss!

Reply:
After our discussion earlier, I changed this behaviour to exclude invalid fields, rather than to only include specified common fields. Also note parallel work being done on GHG in #179 

I think we can break out `pystac` integration to a separate issue, either to be added as a step in Airflow or as a validator in `veda-backend`.